<?xml version="1.0" encoding="utf-8"?>
<launch>
    <node name="rover6_dronecan" pkg="rover6_dronecan" type="rover6_dronecan_node.py" output="screen">
        <!-- CAN bus/dronecan parameters -->
        <param name="can_interface" type="string" value="vcan0" />
        <param name="node_id" type="int" value="124" />
        <param name="bitrate" type="int" value="1000000" />
        <param name="enable_nodeid_server" type="boolean" value="True" />
        <param name="controller_node_id" type="int" value="97" />
        <!-- End of CAN bus/dronecan parameters -->
        <!-- BatteryInfo parameters -->
        <!-- current_offset will be added to BatteryInfo current -->
        <param name="current_offset" type="double" value="0.0" />
        <!-- voltage_offset will be added to BatteryInfo voltage -->
        <param name="voltage_offset" type="double" value="0.0" />
        <!-- if calculate_percentage is true node will calculate percentage from BatteryInfo voltage-->
        <param name="calculate_percentage" type="boolean" value="true" />
        <!-- cell voltage when empty -->
        <param name="cell_empty" type="double" value="3.3" />
        <!-- cell voltage when full -->
        <param name="cell_full" type="double" value="4.2" />
        <!-- number of S cells in battery -->
        <param name="cell_num" type="int" value="6" />
        <!-- if negative current means charging the battery -->
        <param name="negative_charge" type="boolean" value="true" />
        <!-- report temperature from BatteryInfo -->
        <param name="report_temperature" type="boolean" value="false" />
        <!-- End of BatteryInfo parameters -->
    </node>
</launch>
